src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorsAndIndexer/UnnamedSymbolCompletionProvider.cs

```diff
@@ -0,0 +1,200 @@
+    {
+        /// <summary>
+        /// CompletionItems for indexers/operators should be sorted below other suggestions like methods or properties
+        /// of the type.  We accomplish this by placing a character known to be greater than all other normal identifier
+        /// characters as the start of our item's name. this doesn't affect what we insert though as all derived
+        /// providers have specialized logic for what they need to do.
+        /// </summary> 
+        private const string SortingPrefix = "\uFFFD";
+
+        /// <summary>
+        /// Used to store what sort of unnamed symbol a completion item represents.
+        /// </summary>
+        internal const string KindName = "Kind";
+        internal const string IndexerKindName = "Indexer";
+        internal const string OperatorKindName = "Operator";
+        internal const string ConversionKindName = "Conversion";
+
+        /// <summary>
+        /// Used to store the doc comment for some operators/conversions.  This is because some of them will be
+        /// synthesized, so there will be no symbol we can recover after the fact in <see cref="GetDescriptionAsync"/>.
+        /// </summary>
+        private const string DocumentationCommentXmlName = "DocumentationCommentXml";
+
+        [ImportingConstructor]
+        [System.Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public UnnamedSymbolCompletionProvider()
+        {
+        }
+
+        public override ImmutableHashSet<char> TriggerCharacters => ImmutableHashSet.Create('.');
+
+        public override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+            => text[insertedCharacterPosition] == '.';
+
+        /// <summary>
+        /// We keep operators sorted in a specific order.  We don't want to sort them alphabetically, but instead want
+        /// to keep things like <c>==</c> and <c>!=</c> together.
+        /// </summary>
+        private static string SortText(int sortingGroupIndex, string sortTextSymbolPart)
+            => $"{SortingPrefix}{sortingGroupIndex:000}_{sortTextSymbolPart}";
+
+        /// <summary>
+        /// Gets the dot-like token we're after, and also the start of the expression we'd want to place any text before.
+        /// </summary>
+        private static (SyntaxToken dotLikeToken, int expressionStart) GetDotAndExpressionStart(SyntaxNode root, int position)
+        {
+            var tokenOnLeft = root.FindTokenOnLeftOfPosition(position, includeSkipped: true);
+            var dotToken = tokenOnLeft.GetPreviousTokenIfTouchingWord(position);
+
+            // Has to be a . or a .. token
+            if (!CompletionUtilities.TreatAsDot(dotToken, position - 1))
+                return default;
+
+            // don't want to trigger after a number.  All other cases after dot are ok.
+            if (dotToken.GetPreviousToken().Kind() == SyntaxKind.NumericLiteralToken)
```

Comment:
Would this allow casting 'char' to int?

Reply:
i'm not certain i get your question in relation to this line of code.  can you clarify?